midterm-2023-11-07/code/5227-andrej-arsov/script.ts

```diff
@@ -0,0 +1,105 @@
+interface Oscars {
+    [oscarType: string]: string;
+}
+
+interface Cast {
+    actor: string;
+    character: string;
+}
+
+interface Movie {
+    id: number;
+    title: string;
+    year: number;
+    director: string;
+    genre: string[];
+    plot: string;
+    cast: Cast[];
+    oscars: Oscars;
+    rating: number;
+}
+
+document.addEventListener("DOMContentLoaded", siteCode);
+
+async function siteCode() {
+    const data = await loadData();
+    displayMovies(data as Movie[]);
+}
+
+const loadData = async () => {
+    const dataUri = "https://raw.githubusercontent.com/sweko/uacs-internet-programming-exams/main/midterm-2023-11-07/data/movies.json";
+    const response = await fetch(dataUri);
+
+    if (!response.ok) {
+        throw new Error("The data is not there");
+    }
+
+    const data = await response.json();
+    return data;
+}
+
+const displayMovies = (movies: Movie[]) => {
+    const container = document.getElementById("movie-container")!;
+    for (const movie of movies) {
+        const movieRow = generateMovieRow(movie);
+        container.appendChild(movieRow);
+    }
+}
+
+const generateMovieRow = (movie: Movie) => {
+    const row = document.createElement("div");
+    row.classList.add("movie-table");
+
+    createCell(row, "movie-id", movie.id.toString());
+    createCell(row, "movie-name", movie.title);
+    createCell(row, "movie-director", movie.director);
+    createCell(row, "movie-year", movie.year.toString());
+    createCell(row, "movie-genre", movie.genre.join(", "));
+
+    const plotContent = movie.plot.length > 50 ? movie.plot.substr(0, 50).split(' ').slice(0, -1).join(' ') + '...' : movie.plot;
+    createCell(row, "movie-plot", plotContent);
+
+    const castCell = document.createElement("div");
+    castCell.classList.add("movie-data", "movie-cast");
+
+    const sortedCast = movie.cast.sort((a, b) => a.actor.localeCompare(b.actor));
+
+    for (const actorData of sortedCast) {
```

Comment:
Ok extraction and manipulation of actor data. `ul` would have been better and more manageable, but, ok as well +8